### PR TITLE
ci: require drizzle migration when db schema changes

### PR DESCRIPTION
**Pull request by jansinger**
## Zusammenfassung

Seit #582 wenden deployte Umgebungen **ausschließlich generierte Migrationen** beim Container-Start an (`scripts/docker-migrate.ts`). Eine Änderung an `src/lib/server/db/schema.ts` ohne committete Migration unter `drizzle/` würde in Produktion daher **nie ankommen** — bisher gab es dafür keine technische Absicherung, nur die Regel in `.claude/rules/database.md`.

Dieser PR ergänzt in `ci.yml` einen neuen Job **`migration-check`** (nur bei PRs, per `dorny/paths-filter` gated auf `schema.ts`, `drizzle.config.ts` und `drizzle/**`):

1. **Fast-Fail-Diff-Check:** `git diff --name-only origin/<base>...HEAD` — schlägt mit klarer Fehlermeldung fehl, wenn `schema.ts` oder `drizzle.config.ts` geändert wurde, aber weder `drizzle/*.sql` noch `drizzle/meta/_journal.json` angefasst wurde.
2. **Sync-Verifikation:** `npx drizzle-kit generate` (arbeitet offline gegen die Snapshots in `drizzle/meta/`, keine DB nötig) — schlägt fehl, wenn danach `git status --porcelain drizzle/` nicht leer ist. Fängt auch halb generierte oder inkonsistent handeditierte Migrationen ab, die den Diff-Check passieren würden.

Zusätzlich dokumentiert `.claude/rules/database.md` jetzt, dass die Regel CI-erzwungen ist.

## Verifikation (lokal)

- In-sync-Schema: `drizzle-kit generate` → „No schema changes", `drizzle/` bleibt clean ✅
- Dummy-Spalte in `schema.ts` ohne Migration: generate erzeugt `0001_*.sql` + Journal-/Snapshot-Änderungen → Check schlägt fehl ✅
- Grep-Logik gegen 7 Edge-Cases getestet (nur Schema, Schema+SQL, nur Journal, unrelated, nur Config, ähnlicher Dateiname, `drizzle/`-Datei ohne SQL) ✅
- YAML-Syntax validiert ✅

**Kein Unit-Test:** CI-Workflow-Konfiguration (Ausnahme laut `testing.md`), im Commit dokumentiert.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -128,7 +134,81 @@ jobs:
...
+        run: |
+          BASE="origin/${{ github.base_ref }}"
+          CHANGED=$(git diff --name-only "$BASE"...HEAD)
+
+          if echo "$CHANGED" | grep -qxE 'src/lib/server/db/schema\.ts|drizzle\.config\.ts'; then
+            if ! echo "$CHANGED" | grep -qE '^drizzle/.*\.sql$|^drizzle/meta/_journal\.json$'; then
+              echo "::error file=src/lib/server/db/schema.ts::Schema geändert, aber keine Migration committet."
+              echo ""
+              echo "❌ src/lib/server/db/schema.ts (oder drizzle.config.ts) wurde geändert,"
+              echo "   aber der PR enthält keine neue Migrationsdatei unter drizzle/."
```
[the remark this answers is not in the document]
Comment (jansinger) [reply .github/workflows/ci.yml]: Behoben in a1634e0: Die geänderten Dateien werden jetzt ermittelt — jede bekommt eine eigene `::error file=…`-Annotation und die Meldung listet die betroffenen Datei(en) statt fest auf schema.ts zu verweisen.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
